common-test/src/main/java/io/customer/commontest/core/BaseTest.kt

```diff
@@ -1,137 +1,54 @@
+ *      └─ [RobolectricTest] (for all JVM tests using Robolectric with JUnit4)
  */
 abstract class BaseTest {
-
-    protected val siteId: String
-        get() = "test-site-id"
-
-    protected val context: Context
-        get() = InstrumentationRegistry.getInstrumentation().targetContext
-
-    protected val application: Application
-        get() = ApplicationProvider.getApplicationContext()
-
-    protected lateinit var cioConfig: CustomerIOConfig
-
-    protected lateinit var deviceStore: DeviceStore
-    protected lateinit var dispatchersProviderStub: DispatchersProviderStub
-
-    protected lateinit var staticDIComponent: CustomerIOStaticComponent
-
-    protected lateinit var di: CustomerIOComponent
-    protected val jsonAdapter: JsonAdapter
-        get() = di.jsonAdapter
-
-    protected lateinit var mockWebServer: MockWebServer
-
-    // convenient method for test functions to test a failed HTTP request
-    protected fun getHttpError(code: Int, body: String = "{}"): HttpException {
-        return HttpException(Response.error<String>(code, body.toResponseBody()))
+    protected open fun setup(testConfig: TestConfig = testConfigurationDefault { }) {
+        testConfig.configureSDKComponent()
+        registerAndroidSDKComponent(testConfig)
     }
 
-    protected fun createConfig(
-        client: Client = Client.fromRawValue(source = "AndroidTest", sdkVersion = "1.0.0-alpha.6"),
-        siteId: String = this.siteId,
-        apiKey: String = "xyz",
-        region: Region = Region.EU,
-        timeout: Long = 100,
-        autoTrackScreenViews: Boolean = true,
-        autoTrackDeviceAttributes: Boolean = true,
-        backgroundQueueMinNumberOfTasks: Int = 10,
-        backgroundQueueSecondsDelay: Double = 30.0,
-        backgroundQueueTaskExpiredSeconds: Double = Seconds.fromDays(3).value,
-        logLevel: CioLogLevel = CioLogLevel.DEBUG,
-        trackingApiUrl: String? = null,
-        modules: Map<String, CustomerIOModule<*>> = emptyMap()
-    ) = CustomerIOConfig(
-        client = client,
-        siteId = siteId,
-        apiKey = apiKey,
-        region = region,
-        timeout = timeout,
-        autoTrackScreenViews = autoTrackScreenViews,
-        autoTrackDeviceAttributes = autoTrackDeviceAttributes,
-        backgroundQueueMinNumberOfTasks = backgroundQueueMinNumberOfTasks,
-        backgroundQueueSecondsDelay = backgroundQueueSecondsDelay,
-        backgroundQueueTaskExpiredSeconds = backgroundQueueTaskExpiredSeconds,
-        logLevel = logLevel,
-        trackingApiUrl = trackingApiUrl,
-        modules = modules
-    )
-
-    // override in test class to override SDK config for all test functions in the class.
-    protected open fun setupConfig() = createConfig()
-
-    // default @Before for tests using a default SDK config set
-    @Before
-    open fun setup() {
-        setup(cioConfig = setupConfig())
+    protected open fun teardown() {
+        clearAllComponents()
```

Comment:
might as well just call from the method, why do we need another method? 

```
SDKComponent.reset()
clearAllMocks()
```

Reply:
Yeah, removed